core/src/main/java/io/opentelemetry/android/internal/session/SessionManagerImpl.kt

```diff
@@ -57,6 +61,10 @@ internal class SessionManagerImpl(
         if (newSession != session) {
             val previousSession = session
             session = newSession
+            Context
+                .current()
```

Comment:
Considering that a context can change when making a span current, I'm wondering what would happen when this code is called within a nested span, for example:

```
Span A
├─ Span B <-- This is the current context
Span C
```

In that example, the new session will get stored within the Span B context, which will end before the Span C is created. Will the "Span C" have access to the new session? Given that it's not the same context?

Reply:
I was pretty sure that by setting this early initially we would be on the root context, but I think there's nothing to ensure this, and so it's risky. Good catch.